File: backend/Models/Comment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;


namespace backend.Models
{
    [PrimaryKey(nameof(CommentID))]
    public class Comment
    {
        public int CommentID { get; set; }
        [Required]
        public int LocationID { get; set; }
        [Required]
        public int UserID { get; set; }
        [Required]
        [Column("Comment")]
        public string UserComment { get; set; }
        public int? ParentCommentID { get; set; }
        [ForeignKey("ParentCommentID")]
        public ICollection<Comment>? Replies { get; set; }
        public User? User { get; set; }
        [NotMapped]
        public string? Username { get; set; }
    }
}

File: frontend/src/components/models/Comment.ts
class CommentDTO {
    public commentID: number;
    public userID: number;
    public locationID: number;
    public userComment: string;
    public parentCommentID?: number;
    public replies?: CommentDTO[];
    public username?: string
    
    constructor(commentID: number, userID: number, locationID: number, userComment: string, parentCommentID?: number, replies?: CommentDTO[], username?: string) {
        this.commentID = commentID;
        this.userID = userID;
        this.locationID = locationID;
        this.userComment = userComment;
        this.parentCommentID = parentCommentID;
        this.replies = replies;
        this.username = username
    }
}

export default CommentDTO